### ISSUE TITLE
False positive for DMI_RANDOM_USED_ONLY_ONCE with static Random object and lazy init

### ISSUE THREAD

**Issue by guillerodriguez** [spotbugs/spotbugs#2128]
Simplified test case that results in a false positive:

```java
import java.util.Random;

public class RandomTest
{
    private static Random random;

    public static int generateOffset(int negativeMin, int positiveMax)
    {
    	if (random == null)
            random = new Random();

        // [ -negativeMin .. +positiveMax ]
    	return random.nextInt(positiveMax + negativeMin + 1) - negativeMin;
    }
}
```

Reproducible with 4.7.1 and effort = max.
